admin/src/js/controllers/targets-edit.js

```diff
@@ -134,16 +146,26 @@ angular.module('controllers').controller('TargetsEditCtrl',
       $scope.status = 'Submitting';
 
       Settings()
-        .then(updateItem)
-        .then(UpdateSettings)
-        .then(function() {
-          $scope.saving = false;
-          $scope.status = 'Saved';
-        })
-        .catch(function(err) {
-          $log.error('Error updating settings', err);
-          $scope.saving = false;
-          $scope.status = 'Save failed';
+        .then(isTargetUnique)
+        .then(results => {
+          if (!results.unique) {
```

Comment:
`isTargetUnique` doesn't need to be async so it'd be better to make it a pure function. It'll run faster, be easier to read. I know chaining promises is beautiful but sync functions are beautiful in their own way! How about this...

```js
Settings()
  .then(settings => {
    if (!isTargetUnique(settings)) {
      // do error handling
      return;
    }
    return updateItem(settings)
      .then ...
  });
```

While we're at it... updateItem and deleteItem don't need to be async either! If you felt like cleaning them up and saving a few ticks that'd be nice too.